rocky/reports/report_types/vulnerability_report/report.py

```diff
@@ -0,0 +1,129 @@
+
+        for ip in ips:
+            findings_ip_port = self.octopoes_api_connector.query(
+                "IPAddress.<address[is IPPort].<ooi[is Finding]", valid_time, ip.reference
+            )
+
+            findings_software = self.octopoes_api_connector.query(
+                "IPAddress.<address [is ResolvedHostname]"
+                ".hostname.<netloc [is HostnameHTTPURL]>.<ooi [is SoftwareInstance].<ooi [is Finding]",
+                valid_time,
+                ip.reference,
+            )
+
+            for finding in findings_ip_port:
+                findings.append(finding)
+            for finding in findings_software:
+                findings.append(finding)
+
+            for finding in findings:
+                finding_type = self.octopoes_api_connector.query(
+                    "Finding.finding_type", valid_time, Reference.from_str(finding)
+                )[0]
+                if finding_type.reference.class_type != KATFindingType:
+                    finding_types.append(finding_type)
+                    aggregated_findings.append(finding)
+
+        return {"finding_types": finding_types, "findings": aggregated_findings}
+
+    def generate_data(self, input_ooi: str, valid_time: datetime) -> Dict[str, Dict[str, Any]]:
+        findings = self.get_findings(input_ooi, valid_time)
+
+        vulnerabilities = {}
+        occurrences = Counter(findings["finding_types"])
+        critical = RiskLevelSeverity.CRITICAL.value
+        total_criticals = 0
+        first_seen = "-"
+        last_seen = "-"
+        terms = list(occurrences)
+        recommendations = []
+
+        finding_types = list(set(findings["finding_types"]))
+
+        for finding_type in finding_types:
+            filtered_findings = {}
+
+            for finding in findings["findings"]:
+                if finding.finding_type.tokenized.id == finding_type.id:
+                    time_history = self.get_finding_valid_time_history(finding.primary_key)
+
+                    if time_history:
+                        last_seen = time_history[0]
+
+                    filtered_findings[finding.primary_key] = {
+                        _("Source"): "-",
+                        _("First seen"): first_seen,
+                        _("Last seen"): last_seen,
+                        _("Evidence"): "-",
+                    }
+
+                    vulnerabilities[finding_type] = {
+                        "cvss": {
+                            "class": str(finding_type.risk_severity.value).lower(),
+                            "risk_level": str(finding_type.risk_severity.value).title()
+                            + " "
+                            + str(finding_type.risk_score),
+                        },
+                        "occurrences": occurrences[finding_type],
+                        "description": finding_type.description or "-",
+                        "advice": finding_type.recommendation or "-",
```

Comment:
It is probably better to leave it up to the template to render `-` on falsy values